src/ruby/spec/generic/interceptors_spec.rb

```diff
@@ -0,0 +1,221 @@
+# Copyright 2015 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+require 'spec_helper'
+
+describe 'Interceptors' do
+  describe 'interception' do
+    let(:interceptor) { TestServerInterceptor.new }
+    let(:request) { EchoMsg.new }
+    let(:service) { EchoService }
+
+    before(:each) do
+      build_rpc_server
+    end
+
+    context 'when an interceptor is added' do
+      before(:each) do
+        @server.interceptors[:test] = interceptor
+      end
+
+      it 'should be called', server: true do
+        expect(interceptor).to receive(:call).once.and_call_original
+
+        run_services_on_server(@server, services: [service]) do
+          stub = EchoStub.new(@host, :this_channel_is_insecure, **@client_opts)
+          expect(stub.an_rpc(request)).to be_a(EchoMsg)
+        end
+      end
+    end
+
+    context 'when multiple interceptors are added' do
+      let(:interceptor2) { TestServerInterceptor.new }
+      let(:interceptor3) { TestServerInterceptor.new }
+
+      before(:each) do
+        @server.interceptors[:test] = interceptor
```

Comment:
Yes, and it provides a clean interface to the registry of interceptors; you can easily deregister one as if it were a Hash: `server.interceptors[:test] = nil`, or change it by assigning a new interceptor to the same key. I took this approach with the registry as a Hash is generally the easiest way to ensure uniqueness for interceptors within it, while providing a interface that Ruby developers would be automatically familiar with. Happy to change if we feel that's not a good idea.